### DISCUSSION TITLE
support windows7?

### DISCUSSION THREAD

**Discussion by softboy99** [unoplatform/uno#10380] [Q&A]
support windows7?

**Comment by FrozDark** [comment]
By Skia backed (WPF, GTK)

**Reply by lavahasif** [reply]
could you explain.i just create a hello world application in uno (wpf) .and the bin file copy to the windows 7 not working.even not showing error

**Comment by lavahasif** [comment]
> could you explain.i just create a hello world application in uno (wpf) .and the bin file copy to the windows 7 not working.even not showing error

i build using .net 7

**Reply by jeromelaban** [reply]
You likely need to install the [.NET 7 runtime](https://dotnet.microsoft.com/en-us/download/dotnet/7.0). If it still does not work, compile the in debug (Ensuring that `OutputType` is set to `Exe`) and launch the app from a command line prompt.

**Reply by lavahasif** [reply]
WINDOWS7 NOT SUPPORT .NET7 THEN WHAT I DO?

**Reply by jeromelaban** [reply]
Windows 7 is out of support so Microsoft won't mention that it's support, yet you should still be able to install it. You can also try with .NET 6.

**Reply by lavahasif** [reply]
i installed .net 6 , 32 bit and installed perfectly.after rebuild the project in .net 6   and copy file to the windows7. it didnt work. and not showing any errory in event viewer and also no popup

**Reply by jeromelaban** [reply]
Try compiling the in debug (Ensuring that OutputType is set to Exe) and launch the app from a command line prompt.

**Reply by lavahasif** [reply]
i just tried windows built in wpf .net6  template and publish in selfcontained.it worked.framework dependent not working
.thne i tried uno self contained it showing error.
i could'nt under stand this can you clarify ##Ensuring that OutputType is set to Exe)

**Reply by lavahasif** [reply]
> i just tried windows built in wpf .net6 template and publish in selfcontained.it worked.framework dependent not working .thne i tried uno self contained it showing error. i could'nt under stand this can you clarify ##Ensuring that OutputType is set to Exe)

Error showing on self container publish

02/Dec/2022 10:21
System.AggregateException: One or more errors occurred. ---> Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. We were unable to determine the cause of the error. Check the output log for more details. 
   --- End of inner exception stack trace ---
---> (Inner Exception #0) Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. We were unable to determine the cause of the error. Check the output log for more details. <---

Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. We were unable to determine the cause of the error. Check the output log for more details. 

===================

**Reply by lavahasif** [reply]
i just tried windows built in wpf .net6 template and publish in selfcontained.it worked in windows7.framework dependent not working
.thne i tried uno self contained it showing error.
i could'nt under stand this can you clarify ##Ensuring that OutputType is set to Exe)

**Reply by jeromelaban** [reply]
You can change the output type here:

https://github.com/unoplatform/uno/blob/15a6fbccfcd94407bb212eff8e90f51218861703/src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Skia.WPF/UnoQuickStart.Skia.WPF.csproj#L4-L5

Set it to Exe in Release so you can see the app's logs.

**Reply by lavahasif** [reply]
still showing same 
no apps log showing debug and release  also eventviwer too


![notworkingwindows7](https://user-images.githubusercontent.com/22430922/206246258-cb10772c-09d2-4007-be46-16bb8dcc39e1.gif)

**Reply by lavahasif** [reply]
i just write direct wpf it work in dotnet6
![as](https://user-images.githubusercontent.com/22430922/206249248-5ebf464a-8352-4752-84d7-c1c715b9a876.gif)

**Reply by jeromelaban** [reply]
When an app is built with `OutputType` set to `Exe`, you need to run the app using a command line shell. Type `CMD` in the "run" window, then run the app from there.
